### PR TITLE
#21 ssl support

### PR DESCRIPTION
**Pull request by neo7337**

### CODE REVIEW

File: clients/rest/rest.go [hunk 1]
```diff
@@ -123,6 +126,46 @@ func (c *Client) SetProxy(proxyUrl, user, password string) (err error) {
 	return
 }
 
+func (c *Client) SetCACerts(caFilePath string) (*Client, error) {
```
Comment (nandagopalan) [inline clients/rest/rest.go:129]: May be this can take more than one ca to add multiple Intermediate CAs at once

Comment (nandagopalan) [inline clients/rest/rest.go:129]: May be this can take more than one ca to add multiple Intermediate CAs at once

File: clients/rest/rest.go [hunk 2]
```diff
@@ -123,6 +126,46 @@ func (c *Client) SetProxy(proxyUrl, user, password string) (err error) {
...
+		return nil, err
+	}
+	for _, v := range certs {
+		conf.Certificates = append(conf.Certificates, v)
+	}
+	c.setSSL(conf)
+	return c, nil
+}
+
+func (c *Client) setSSL(conf *tls.Config) {
```
Comment (nandagopalan) [inline clients/rest/rest.go:160]: This and the SSlVerify function will override the *tls.Config on the client Struct.

Need to add nil check and update if present

File: clients/rest/rest.go [hunk 3]
```diff
@@ -214,3 +257,21 @@ func (c *Client) Close() (err error) {
 	c.httpClient.CloseIdleConnections()
 	return
 }
+
+func (c *Client) setTlSConfig() (*tls.Config, error) {
```
Comment (nandagopalan) [inline clients/rest/rest.go:261]: This looks like a getMethod but also sets its.

May be we can simplify with

1 Just keep the TLS config separate 
2 Call a prepareTransport func in execute to just set at once

**Review by nandagopalan — CHANGES_REQUESTED** [review]

File: clients/rest/rest.go (hunk 1 above)
Comment (neo7337) [reply clients/rest/rest.go:129]: added!

File: clients/rest/rest.go (hunk 2 above)
Comment (neo7337) [reply clients/rest/rest.go:160]: updated

**Review by nandagopalan — APPROVED** [review]
Approved

**Pull request merged**